### PR TITLE
Add release notes for 0.263

### PR DESCRIPTION
**Pull request by tdcmeehan**
# Missing Release Notes

# Extracted Release Notes
- #16529 (Author: Pranjal Shankhdhar): Support distinct types
  - Support distinct user defined types.
- #16568 (Author: Ajay George): Add LimitWithoutOrderByChecker for filtering non-deterministic types
  - Improve Verifier by skipping the non-deterministic queries which have a ``LIMIT`` sub clause without ``ORDER BY``.
- #16669 (Author: Arjun Gupta): Add support to perform heapDump on exceeded memory limit failures
  - Add a new session property `heap_dump_on_exceeded_memory_limit_enabled` to enable heapdump on exceeded memory failures. The heapdump file directory can be provided using `exceeded_memory_limit_heap_dump_file_directory` session property.
- #16747 (Author: James Petty): Fix distinct aggregation input filtering when input mask has nulls
  - Fix a potential correctness bug in distinct aggregations when combined with a filtering mask containing null values.
- #16766 (Author: Sreeni Viswanadha): Implement key-based sampling in the planner
  - Added Smart Sampling feature that samples the tables at query (planning) time. This helps in cases when the user is just exploring the data, testing a pipeline or looking for trends. We sample all the tables (and only tables) in such a way that all "key-based" operations like JOIN/GROUP BY/window functions etc will be complete for the sampled keys.
- #16788 (Author: Arjun Gupta): Buffer and spill only required channels for distinct aggregates
  - Improve spilling CPU and memory efficiency for distinct and order by aggregations.

# All Commits
- ae771008b3e1c9763ff78b4df539d5d319a041ba Implement key-based sampling using a deterministic/stable sampling function (Sreeni Viswanadha)
- 0b07436d5eb98f4a5985cd5462307e451ff1349e Refactor SqlTypeBytecodeExpression (Pranjal Shankhdhar)
- d02882d77add4a26c5afdcb02ef0df36ecc44ea0 Add support to write codegen scalar functions (Pranjal Shankhdhar)
- bfdd18570e46235e9c5b0f2a25bc4bc01a61d597 Remove state/PageBuilder from scalar map/array operators (Sreeni Viswanadha)
- 4fbc2b170d7fe7fe7a6c947c19d2ce7bfc63ac3b Fix coordinator - resource manager proxy communication (Swapnil Tailor)
- f5fe69e30a0c9ecc22e4ce108ec4260d2054ffff Ensure correct ordering of heartbeats in resource manager (Tim Meehan)
- 8d870290a3427e708fe101a77b5ace626411c6fa Config for DwrfStripeCache in OrcReader (Arunachalam Thirupathi)
- c970812be57b7c228814d55c65df44e0770a778c Disable flaky tests (abhiseksaikia)
- 809dfe8c723fecd6ec814a728cd782a9e3bb8554 Add stats for materialized view query rewrite (Zac Wen)
- d044348915528d086a6c9f296b484e12d6f77221 Disabling flaky TestQueues test (Swapnil Tailor)
- a027b23635a3c4c8a390610f53fefd6925973f5b Check for RLE masks in DistinctingAccumulator (James Petty)
- 9f746bba20e52bed3bcb05fa58a220e0b02d171d Test RLE masks on distinct aggregations (James Petty)
- fb4ef596c272913e2e2060dca5490cefea803fc6 Improve AccumulatorCompiler generated code (James Petty)
- 5dee73fef84bf6a48acf31d55a66ff8575970aa5 Categorize metastore user errors correctly (Nikhil Collooru)
- 81ff17a07582effb62f706ff87eafab7f4885462 Buffer and spill only required channels for distinct aggregates (Arjun Gupta)
- 589b0a52a9a8e079d78aa484488309c1dd37c634 Use block.mayHaveNull() to skip isNull checks in some block serializers (James Petty)
- b2e62d9b96bec9c775aa73660c6e3d683c7fd5a5 Deserialize primitive arrays with no null positions via direct copy (James Petty)
- 645ef77779ccd6690f509032ac8963cf4200767d Avoid all-false valueIsNull in Row and Array block deserializer (James Petty)
- 2d4859ed8c314a97a0d23c00a9b70910b062c3f9 Add LimitWithoutOrderByChecker for filtering non-deterministic types (Ajay George)
- 03efeeb8af4fa21c658ed6b825386f723e178da1 Fix parameter name in StripeMetadataSourceFactory (Arunachalam Thirupathi)
- 727fac66bf71fd909b7bc06a95bf898331f748ce Fix dictionary row group size test (Arunachalam Thirupathi)
- 72339244b03c87ee1a3288967e198d5d2544c465 Convert StripeMetadataSource to Factory pattern (Arunachalam Thirupathi)
- 6cb9a8d3a6ca3caeefb5c542821a6cf1de403cd7 Fix incorrect dwrf file raw size (Arunachalam Thirupathi)
- 7424533b4a1bc1863e3eee28a2e07b7f8f1d20d3 Support multiple candidates for materialized view (Zac Wen)
- cfc45b786c21dd5a5daa825c803b8736e869c348 Revert "Record operator output before addPage in LocalExchangeSink" (Neerad Somanchi)
- e8093751c6576b9c9740b1aef9868f4e8475f9d7 Revert "Ensure blocks are fully loaded by when enforcing output layouts" (Neerad Somanchi)
- cae41c20b796728d9efcc37cc785e47cc66784ef Revert "Improve PartitioningExchanger performance" (Neerad Somanchi)
- 0056030cc0386e75c6f8112c2112145c4baefc48 Config for loose memory accounting to run queries without OOM (SOURAV PAL)
- 0fb160be7286b932ba15c695fe5375d28949bb96 Fix Resource Group Distributed Queuing (Swapnil Tailor)
- 84172e599cd3be7c53666933a552d4c3eea752b0 Config for DwrfStripeCache in OrcWriter (Arunachalam Thirupathi)
- 74257e19b98a8d66b87ad96f276d17bdeb5e4161 Handle pushdown predicates in MetadataQueryOptimizer (Zhan Yuan)
- e389528e7f919f4e48d7da4cef3c8f69be2d583d Include all tables in the column mapping of materialized views (Ge Gao)
- 87512317beea3f8cce4b5e26c75be96caa63bf16 Support GROUP BY in materialized view rewriting (Zac Wen)
- 2d0e848bc7fa4f5eb12c87b8d57c4faf5ace0941 Process expressions inside IN and IF for materialized view rewrite (Zac Wen)
- 1098248df0ef67d8f324ec373402ab077901b2b7 Support distinct types (Pranjal Shankhdhar)
- 8ebf7d99061399ea11ae94d0bd2254911e142687 Enabling flaky TestQueues test (Swapnil Tailor)
- 70a81bcc59e3f88eb83af99a290efef2c2a9ffa2 Fix distinct aggretation input filtering when input mask has nulls (James Petty)
- fb5b8b578b9e0570d66392b7d8c577fe857747ef Enabling Flaky Tests As they run reliably now (Swapnil Tailor)
- ef141fe65fddf93878773cbada9c55fc4bae9013 Support for query heartbeat from coordinator that is shutting down (Abhisek Gautam Saikia)
- 80810d51d7f09c78ae5f212f1e6f71014dca5c5a Revert "Handle fully filter pushdown for metadata query rewrite" (Zhan Yuan)
- ed2a648b46225860260180aad47a0a7e853f366f Add support to perform heapDump on exceeded memory limit failures (Arjun Gupta)

### CODE REVIEW

File: presto-docs/src/main/sphinx/release/release-0.263.rst [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+**Details**
+===========
+
+General Changes
+_______________
+* Fix a correctness bug in distinct aggregations when combined with a filtering mask containing null values.
+* Improve spilling CPU and memory efficiency for distinct and order by aggregations.
+* Add a new session property ``heap_dump_on_exceeded_memory_limit_enabled`` to enable heapdump on exceeded memory failures. The heapdump file directory can be provided using ``exceeded_memory_limit_heap_dump_file_directory`` session property.
+* Added Smart Sampling feature that samples the tables at query (planning) time. This helps in cases when the user is just exploring the data, testing a pipeline or looking for trends. We sample all the tables (and only tables) in such a way that all "key-based" operations like JOIN/GROUP BY/window functions etc will be complete for the sampled keys.
```
Comment (tdcmeehan) [inline presto-docs/src/main/sphinx/release/release-0.263.rst:13]: @kaikalur Any config/session properties?  Can you please reference those properties' documentation and make this less verbose?

File: presto-docs/src/main/sphinx/release/release-0.263.rst [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+Release 0.263
+=============
+
+**Details**
+===========
+
+General Changes
+_______________
+* Fix a correctness bug in distinct aggregations when combined with a filtering mask containing null values.
+* Improve spilling CPU and memory efficiency for distinct and order by aggregations.
```
Comment (ajaygeorge) [inline presto-docs/src/main/sphinx/release/release-0.263.rst:11]: \``DISTINCT\`` and  \``ORDER BY``

File: presto-docs/src/main/sphinx/release/release-0.263.rst [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+* Add a new session property ``heap_dump_on_exceeded_memory_limit_enabled`` to enable heapdump on exceeded memory failures. The heapdump file directory can be provided using ``exceeded_memory_limit_heap_dump_file_directory`` session property.
+* Added Smart Sampling feature that samples the tables at query (planning) time. This helps in cases when the user is just exploring the data, testing a pipeline or looking for trends. We sample all the tables (and only tables) in such a way that all "key-based" operations like JOIN/GROUP BY/window functions etc will be complete for the sampled keys.
+* Support distinct user defined types.
```
Comment (ajaygeorge) [inline presto-docs/src/main/sphinx/release/release-0.263.rst:14]: seems a little skinny on details. Can we add some more clarity on what this is.

File: presto-docs/src/main/sphinx/release/release-0.263.rst (hunk 1 above)
Comment (ajaygeorge) [reply presto-docs/src/main/sphinx/release/release-0.263.rst:13]: just say `Add` instead of `Added` .
like  \``JOIN\`` /  \``GROUP BY``

Comment (kaikalur) [reply presto-docs/src/main/sphinx/release/release-0.263.rst:13]: * Added Smart Sampling feature that samples tables at query time. This feature can be enabled using session property ``key_based_sampling_enabled=true``. Additionally, sampling rate and sampling function name can be provided by sessions properties ``key_based_sampling_percentage`` and ``key_based_sampling_function`` respectively.

File: presto-docs/src/main/sphinx/release/release-0.263.rst [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+**Details**
+===========
+
+General Changes
+_______________
+* Fix a correctness bug in distinct aggregations when combined with a filtering mask containing null values.
+* Improve spilling CPU and memory efficiency for ``DISTINCT`` and ``ORDER BY`` aggregations.
+* Add a new session property ``heap_dump_on_exceeded_memory_limit_enabled`` to enable heapdump on exceeded memory failures. The heapdump file directory can be provided using ``exceeded_memory_limit_heap_dump_file_directory`` session property.
+* Added key-based sampling feature that samples tables at query time. This feature can be enabled using session property ``key_based_sampling_enabled=true``. Additionally, default sampling rate and sampling function name can be overridden by sessions properties ``key_based_sampling_percentage`` and ``key_based_sampling_function`` respectively.
```
Comment (ajaygeorge) [inline presto-docs/src/main/sphinx/release/release-0.263.rst:13]: Added -> Add

**Review by ajaygeorge — APPROVED** [review]

**Review by highker — APPROVED** [review]

File: presto-docs/src/main/sphinx/release/release-0.263.rst [hunk 5]
```diff
@@ -0,0 +1,23 @@
+=============
+Release 0.263
+=============
+
+**Details**
+===========
+
+General Changes
+_______________
+* Fix a correctness bug in distinct aggregations when combined with a filtering mask containing null values.
```
Comment (mayankgarg1990) [inline presto-docs/src/main/sphinx/release/release-0.263.rst:10]: this is too implementation specific. What high level use cases does it effect ? Some example query shapes ?

File: presto-docs/src/main/sphinx/release/release-0.263.rst [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+* Improve spilling CPU and memory efficiency for ``DISTINCT`` and ``ORDER BY`` aggregations.
+* Add a new session property ``heap_dump_on_exceeded_memory_limit_enabled`` to enable heapdump on exceeded memory failures. The heapdump file directory can be provided using ``exceeded_memory_limit_heap_dump_file_directory`` session property.
```
Comment (mayankgarg1990) [inline presto-docs/src/main/sphinx/release/release-0.263.rst:12]: The session property is not the feature rather the functionality is. Lets rephrase this as - 

```
Add support to create a heapdump when we exceed local memory limits for a query. This can be enabled with the ``heap....`` session property and the directory to create the heapdump can be set by ``....`` session property.
```

File: presto-docs/src/main/sphinx/release/release-0.263.rst [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+* Add key-based sampling feature that samples tables at query time. This feature can be enabled using session property ``key_based_sampling_enabled=true``. Additionally, default sampling rate and sampling function name can be overridden by sessions properties ``key_based_sampling_percentage`` and ``key_based_sampling_function`` respectively.
+* Support distinct user defined types with ``CREATE TYPE`` syntax.
```
Comment (mayankgarg1990) [inline presto-docs/src/main/sphinx/release/release-0.263.rst:14]: Is this a new language feature ? If yes, we should create documentation and link to the documentation.

File: presto-docs/src/main/sphinx/release/release-0.263.rst [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+Verifier Changes
+________________
+* Improve Verifier by skipping non-deterministic queries which have a ``LIMIT`` sub clause without ``ORDER BY``.
```
Comment (mayankgarg1990) [inline presto-docs/src/main/sphinx/release/release-0.263.rst:18]: `Improve false-positives` maybe ?

**Review by mayankgarg1990 — CHANGES_REQUESTED** [review]

File: presto-docs/src/main/sphinx/release/release-0.263.rst (hunk 5 above)
Comment (pettyjamesm) [reply presto-docs/src/main/sphinx/release/release-0.263.rst:10]: I think this is a little tricky because the bug fixed would only produce incorrect results in the presence of some other behavior that would create boolean mask column with positions such that `BOOLEAN.getBoolean(block, position)` would return `true` where `block.isNull(position)` would _also_ return `true` (ie: null positions are not filtered but should have been, because the value present at that position happened to be non-zero and no null check was performed).

As far as I know, no behaviors in Presto that I'm aware of were creating mask columns with that property, but I can't be 100% sure that no connector or reader might have done something internally that could create those blocks. As such, we could either:
- Remove the release notes item because we do not currently know of any specific scenarios that would have triggered the bug
- Change the release notes to something like:
>* Fix a _potential_ correctness bug for distinct aggregations with filter expressions containing null values
